File: orderbook-backend/MarketData/src/main/java/com/ensi/orderbook/models/OrderBuy.java
package com.ensi.orderbook.models;

import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
@Entity
@Setter
@Getter
@Table(name = "order_buy")
@NoArgsConstructor

public class OrderBuy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private OrderType orderType;

    private int quantity;
    private Double price;

    @ManyToOne
    @JoinColumn(name = "trader_id")
    private Trader trader;
    @ManyToOne
    private Stock stock;
    @ManyToOne
    private OrderBook orderBook;  // Ajout de l'attribut orderBook


    @Enumerated(EnumType.STRING)
    @ColumnDefault("'OUVERT'")
    private OrderStatus status;

    public enum OrderType{
        ACHAT,VENTE
    }
    public enum OrderStatus {
        OUVERT,REMPLI,ANNULE
    }
}

File: orderbook_frontend/src/app/models/order-buy.ts
export class OrderBuy {
    id: number;
    orderType: OrderType;
    quantity: number;
    price: number;
    //trader: Trader;
    stock: {id:number};
    orderBook: {id:number};
    status: OrderStatus;
  
    constructor(
      id: number,
      orderType: OrderType,
      quantity: number,
      price: number,
      //trader: Trader,
      stock: {id:number},
      orderBook: {id:number},
      status: OrderStatus
    ) {
      this.id = id;
      this.orderType = orderType;
      this.quantity = quantity;
      this.price = price;
      //this.trader = trader;
      this.stock = stock;
      this.orderBook = orderBook;
      this.status = status;
    }
  }
  
  export enum OrderType {
    ACHAT = 'ACHAT',
    VENTE = 'VENTE',
  }
  
  export enum OrderStatus {
    OUVERT = 'OUVERT',
    REMPLI = 'REMPLI',
    ANNULE = 'ANNULE',
  }
  